aptos-move/aptos-transaction-simulation/src/state_store.rs

```diff
@@ -0,0 +1,447 @@
+        )
+    }
+    /// Gets an on-chain config.
+    ///
+    /// The Rust type representing the on-chain config must implement the [`OnChainConfig`] trait.
+    fn get_on_chain_config<C>(&self) -> Result<C>
+    where
+        Self: Sized,
+        C: OnChainConfig,
+    {
+        C::fetch_config(self).ok_or_else(|| {
+            anyhow!(
+                "failed to fetch on-chain config: {:?}",
+                std::any::type_name::<C>()
+            )
+        })
+    }
+    /// Modifies an on-chain config in-place.
+    ///
+    /// The Rust type representing the on-chain config must implement the [`OnChainConfig`] trait.
+    fn modify_on_chain_config<C, F>(&self, modify: F) -> Result<()>
+    where
+        Self: Sized,
+        C: OnChainConfig + Serialize,
+        F: FnOnce(&mut C) -> Result<()>,
+    {
+        let mut config = self.get_on_chain_config::<C>()?;
+        modify(&mut config)?;
+        self.set_on_chain_config(&config)
+    }
+
+    /// Stores a compiled Move module, with the module being already serialized into a blob.
+    fn add_module_blob(&self, module_id: &ModuleId, blob: impl Into<Bytes>) -> Result<()> {
+        self.set_state_value(
+            StateKey::module_id(module_id),
+            StateValue::new_legacy(blob.into()),
+        )
+    }
+    /// Stores a compiled Move module.
+    fn add_module(&self, module: &CompiledModule) -> Result<()> {
+        let mut blob = vec![];
+        module.serialize(&mut blob)?;
+
+        self.set_state_value(
+            StateKey::module_id(&module.self_id()),
+            StateValue::new_legacy(blob.into()),
+        )
+    }
+
+    /// Sets the `ChainId` resource that is used to identify the blockchain network.
+    fn set_chain_id(&self, chain_id: ChainId) -> Result<()> {
+        let bytes = bcs::to_bytes(&chain_id)?;
+
+        self.set_state_value(
+            StateKey::resource(ChainId::address(), &ChainId::struct_tag()).unwrap(),
+            StateValue::new_legacy(bytes.into()),
+        )
+    }
+
+    /// Sets the on-chain feature flags.
+    fn set_features(&self, features: Features) -> Result<()> {
+        let bytes = bcs::to_bytes(&features)?;
+
+        self.set_state_value(
+            StateKey::resource(Features::address(), &Features::struct_tag()).unwrap(),
+            StateValue::new_legacy(bytes.into()),
+        )
+    }
+
+    /// Adds the given account_data to the storage.
+    fn add_account_data(&self, account_data: &AccountData) -> Result<()> {
+        let write_set = account_data.to_writeset();
+        self.apply_write_set(&write_set)
+    }
+
+    /// Creates and stores a new account with the given balance and sequence number.
+    fn store_and_fund_account(
+        &self,
+        account: Account,
+        balance: u64,
+        seq_num: u64,
+    ) -> Result<AccountData>
+    where
+        Self: Sized,
+    {
+        let features: Features = self.get_on_chain_config()?;
```

Comment:
Before, we were doing `let features = Features::fetch_config(&self.data_store).unwrap_or_default();` so here we should really make sure features are initialized beforehand? Would it make sense to ensure features always exist in store? Or use defaults?